### DISCUSSION TITLE
Setting up a parametric spider (dynamic base_url and start_urls)

### DISCUSSION THREAD

**Discussion by matteoredaelli** [elixir-crawly/crawly#237] [General]
Hello

How could I develop a parametric spider and run it like

   Crawly.Engine.start_spider(MySpider, base_url, start_urls)

Thanks in advance
Matteo

**Comment by Ziinc** [comment]
Please use spider `init/1` arguments. Reading the [documentation](https://hexdocs.pm/crawly/Crawly.Spider.html) is also advised.
